docs/src/performance_tips.md

```diff
@@ -129,8 +129,5 @@ t = joinpath(dirname(dirname(x)), "test", "performance", "variable_agent_types_s
 include(t)
 ```
 
-The result is that having many types (here 15 different types) makes the code about 5-6 times slower.
+If you want to use many different agent types, you can consider using the [`@multiagent`](@ref) macro which can improve the time of a model run considerably and, in many cases, without incurring almost any additional memory increase.
```

Comment:
these are the relative times calculated with the updated script for 2,3,4,5,10,15 types:

```
[1.3010113741528846, 1.4163006412337629, 4.212580226232111, 5.222170236597688, 7.663690421079443, 12.618665075001266]
```

the relative time of :opt_speed multiagent with 15 subtypes is instead nearly 1 (put in the script too), with :opt_memory it is 3.

We should surely state something about this, removing that was a mistake.

Reply:
in my opinion, given these results, in general I would still suggest to use a `Union` till three types since the impact is small, and you can use dispatch. But above that we should suggest to go with the multiagent macro

Reply:
Okay, yes, that is a great idea, to recommend `Union` for up to three types. Hopefully realistic models don't use more than that.